src/game/World.cpp

```diff
@@ -2816,7 +2816,7 @@ void World::LogTransaction(PlayerTransactionData const& data)
         std::stringstream items;
         for (int i = 0; i < TransactionPart::MAX_TRANSACTION_ITEMS; ++i)
             if (part.itemsEntries[i])
-                items << uint32(part.itemsEntries[i]) << ":" << uint32(part.itemsCount[i]) << ":" << part.itemsGuid[i];
+                items << uint32(part.itemsEntries[i]) << ":" << uint32(part.itemsCount[i]) << ":" << part.itemsGuid[i] << ";";
```

Comment:
This will add a trailing `;`, resulting in an extra empty token after tokenization. You should add some logic to only add ; when necessary. I'd probably add it at the start of the string if it's not the first item.